src/FSharp.Core/local.fs

```diff
@@ -14,6 +14,11 @@ module internal DetailedExceptions =
         let msg = String.Format (format, paramArray)
         raise (new ArgumentException (msg, arg))
 
+    /// takes an argument, a formatting string, a param array to splice into the formatting string
+    let inline invalidArgOutOfRangeFmt (arg:string) (format:string) paramArray =
```

Comment:
This now should be called now in the `invalidArgOutOfRange` down below. Otherwise there will be exception type mismatch happening.

Reply:
I thought about this, but this would be a lot of breaking changes, and I didn't want that to happen

Reply:
So I've raised [this PR](https://github.com/dotnet/fsharp/pull/17328) to have a related discussion, collecting opinions.

I'd say:
- we should either fix this name/implementation inconsistency and use appropriate functions everywhere
- or at least use `invalidArg` when that's what we actually throw, to not bring more misleading code to the codebase

Reply:
Alright, this doesn't apply anymore, resolving.